.editorconfig

```diff
@@ -0,0 +1,12 @@
+root = true
```

Comment:
is it better to exclude this file in the repo so developers could have more flexibility in their IDE?

Reply:
yeah, good point. but I think `editorconfig` can do no harm to developers' editor, it is only a suggestion and universal format for a single project, and it doesn't include any specific coding rules except `tab size`, `trimming whitespace`, and `insert final newline`, nothing else. 

I'd love to suggest we can leave this config here and have a consistent fundamental coding style for us.

Reply:
sounds good.